### PR TITLE
#162949212 Send verification email on register

### PR DESCRIPTION
**Pull request by MbugwaSami**
**What does this PR do?**
- Sends an email to the user with an activation link upon registration

**Description of the Task to be completed**
- generate an account activation link.
- send email on registration.
- activate user account upon click on link
- test verification email and activation functionality.

**How should this be manually tested?**
- Clone the repository: git clone https://github.com/andela/ah-the-unsullied.git
- cd ah-the-unsullied
- git checkout ft-activate-account-162949212
- create a new Gmail account
- [sign in and allow access to less secure apps](https://myaccount.google.com/lesssecureapps)
- create .env on the ah-the-unsullied folder and add the following
```export DATABASE_URL='postgres://USER:PASSWORD@HOST:PORT/NAME'
     export SECRET_KEY="<your_secret_key_here>"
     export HOST_PASSWORD="<Gmail_Password>"
     export HOST_USER="<Gmail_Username>"
     export SMTP_HOST='smtp.gmail.com'
     export EMAIL_PORT='587
```
- Run the following commands 
```python3 -m venv venv
     source .env
     python manage.py migrate
     python manage.py runserver
```
- open Postman and test the app by registering a user with an existing email.
- check mail inbox for an activation link.
   
**Relevant PT story**
#162949212

**Screenshots**
screenshot 2019-01-21 at 12 51 04

### CODE REVIEW

**Review by kelvinwm — DISMISSED** [review]

**Review by Allan690 — CHANGES_REQUESTED** [review]
On pulling your branch and testing locally your views.py file has some minor PEP8 issues on lines 74, and 162(2 lines). urls.py has a line too long issue on line 3. Otherwise, the rest of the work looks good

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def create_superuser(self, username, email, password):
 
       user = self.create_user(username, email, password)
       user.is_superuser = True
-      user.is_staff = True
+      user.is_staff = False
```
Comment (kwanj-k) [inline authors/apps/authentication/models.py:47]: Why did you change the staff status

Comment (nesh-dev) [reply authors/apps/authentication/models.py:47]: It is an error on my part thanks for correcting the changes.

**Comment by nesh-dev** [comment]
> On pulling your branch and testing locally your views.py file has some minor PEP8 issues on lines 74, and 162(2 lines). urls.py has a line too long issue on line 3. Otherwise, the rest of the work looks good.
> 
> For Pycharm users, It is important to note that Pycharm's default line length is 120 characters. You can change that to 79 characters by going to `Files -> Preferences -> Editor -> Codestyle `. This will help you catch a lot of PEP8 issues that had gone under the radar

The lines updated

**Review by Allan690 — DISMISSED** [review]

**Comment by kwanj-k** [comment]
I just tested your feature
I like what you have done with the template and redirect

**Review by kwanj-k — DISMISSED** [review]
Works on Heroku too

**Review by kelvinwm — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]

**Review by sylviawanjiku — DISMISSED** [review]

**Review by sylviawanjiku — DISMISSED** [review]

**Review by kelvinwm — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]

**Review by Allan690 — DISMISSED** [review]

**Review by kelvinwm — DISMISSED** [review]

**Review by kwanj-k — DISMISSED** [review]

File: authors/apps/authentication/views.py [hunk 2]
```diff
@@ -23,6 +35,45 @@ def post(self, request):
...
+        user_pk = user_to_activate.pk
+        user_id = urlsafe_base64_encode(force_bytes(user_pk)).decode()
+
+        # Generate the token
+        token = account_activation_token.make_token(user_to_activate)
+
+        # pass data to the reverse url
+        url = "{}".format(reverse('authentication:activate', kwargs={
+            'pk': user_id,
+            'token': token}
```
Comment (indungu) [inline authors/apps/authentication/views.py]: I believe the curly brace should be in a new line. Refer to the [indentation](https://www.python.org/dev/peps/pep-0008/#indentation) sub-section of the Code layout section of the PEP8 standards.

**Review by indungu — CHANGES_REQUESTED** [review]
Just a slight change... ping me when you're done.

**Review by Allan690 — APPROVED** [review]

**Review by kelvinwm — APPROVED** [review]

**Review by indungu — APPROVED** [review]
